File: spring-boot-tutorial-basics-configuration/src/main/java/com/in28minutes/springboot/tutorial/basics/application/configuration/SpringBootTutorialBasicsConfigurationApplication.java
package com.in28minutes.springboot.tutorial.basics.application.configuration;

import java.util.Arrays;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Profile;

@SpringBootApplication
public class SpringBootTutorialBasicsConfigurationApplication {

    public static void main(String[] args) {
        ApplicationContext applicationContext = SpringApplication.run(SpringBootTutorialBasicsConfigurationApplication.class, args);

        Arrays.stream(applicationContext.getBeanDefinitionNames())
                .forEach(System.out::println);
    }

    @Profile("dev")
    @Bean
    public String devBean() {
        return "dev";
    }

    @Profile("qa")
    @Bean
    public String qaBean() {
        return "qa";
    }

    @Profile("prod")
    @Bean
    public String prodBean() {
        return "prod";
    }
}

File: spring-boot-kotlin-basics-configuration/src/main/kotlin/com/in28minutes/springboot/tutorial/basics/application/configuration/SpringBootTutorialBasicsConfigurationApplication.kt
package com.in28minutes.springboot.tutorial.basics.application.configuration

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Profile

@SpringBootApplication
class SpringBootTutorialBasicsConfigurationApplication {
    @Profile("dev")
    @Bean
    fun devBean(): String {
        return "dev"
    }

    @Profile("qa")
    @Bean
    fun qaBean(): String {
        return "qa"
    }

    @Profile("prod")
    @Bean
    fun prodBean(): String {
        return "prod"
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val applicationContext: ApplicationContext =
                SpringApplication.run(SpringBootTutorialBasicsConfigurationApplication::class.java, *args)
            applicationContext.beanDefinitionNames.iterator()
                .forEach { name -> println(name) }
        }
    }
}